lock-api/src/test/java/com/palantir/lock/client/MultiClientCommitTimestampGetterTest.java

```diff
@@ -0,0 +1,251 @@
+        assertSanityOfResponse(alphaRequestList, ImmutableMap.of(alpha, ImmutableList.of(responseMap.get(alpha))));
+
+        LockWatchEventCache alphaCache = LOCK_WATCH_EVENT_CACHE_MAP_MAP.get(alpha);
+        verify(alphaCache).lastKnownVersion();
+        verify(alphaCache).processGetCommitTimestampsUpdate(any(), any());
+
+        // assert unsuccessful requests were made by client beta
+        assertThat(requestForBeta.result().isDone()).isFalse();
+
+        LockWatchEventCache betaCache = LOCK_WATCH_EVENT_CACHE_MAP_MAP.get(beta);
+        verify(betaCache, times(2)).lastKnownVersion();
+        verify(betaCache, never()).processGetCommitTimestampsUpdate(any(), any());
+    }
+
+    private void setupServiceAndAssertSanityOfResponse(List<BatchElement<NamespacedRequest, Long>> batch) {
+        Map<Namespace, List<GetCommitTimestampsResponse>> expectedResponseMap = new HashMap<>();
+
+        when(timelockService.getCommitTimestamps(any())).thenAnswer(invocation -> {
+            Map<Namespace, GetCommitTimestampsResponse> commitTimestamps =
+                    getCommitTimestampResponse(invocation.getArgument(0));
+            commitTimestamps.forEach((namespace, response) -> {
+                expectedResponseMap
+                        .computeIfAbsent(namespace, _unused -> new ArrayList())
+                        .add(response);
+            });
+            return commitTimestamps;
+        });
+
+        consumer.accept(batch);
+        assertSanityOfResponse(batch, expectedResponseMap);
+    }
+
+    private void assertSanityOfResponse(
+            List<BatchElement<NamespacedRequest, Long>> batch,
+            Map<Namespace, List<GetCommitTimestampsResponse>> expectedResponseMap) {
+        // all requests should be served
+        assertThat(batch.stream().filter(elem -> !elem.result().isDone()).collect(Collectors.toSet()))
+                .isEmpty();
+
+        Map<Namespace, List<Long>> partitionedResponseMap = batch.stream()
+                .collect(groupingBy(
+                        elem -> elem.argument().namespace(),
+                        Collectors.mapping(elem -> Futures.getUnchecked(elem.result()), toList())));
+
+        assertThat(partitionedResponseMap.keySet()).isEqualTo(expectedResponseMap.keySet());
+        assertCorrectnessOfCompletedRequests(expectedResponseMap, partitionedResponseMap);
+    }
+
+    private static void assertCorrectnessOfCompletedRequests(
+            Map<Namespace, List<GetCommitTimestampsResponse>> expectedResponseMap,
+            Map<Namespace, List<Long>> partitionedResponseMap) {
+        KeyedStream.stream(partitionedResponseMap)
+                .forEach((namespace, commitTsList) ->
+                        assertSanityOfServedTimestamps(commitTsList, expectedResponseMap.get(namespace)));
+    }
+
+    private static void assertSanityOfServedTimestamps(
+            List<Long> commitTsList, List<GetCommitTimestampsResponse> commitTimestampsResponses) {
```

Comment:
the 2nd argument should be `expectedCommitTimestampsResponses`. I'd suggest switching them around for consistency with `assertCorrectnessOfCompletedRequests`